format/macho/macho.go

```diff
@@ -0,0 +1,574 @@
+package macho
+
+// https://github.com/aidansteele/osx-abi-macho-file-format-reference
+
+import (
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/bitio"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.MACHO,
+		Description: "Mach-O macOS executable",
+		Groups:      []string{format.PROBE},
+		DecodeFn:    machoDecode,
+	})
+}
+
+//nolint:revive
+const (
+	MH_MAGIC    = 0xfeedface
+	MH_CIGAM    = 0xcefaedfe
+	MH_MAGIC_64 = 0xfeedfacf
+	MH_CIGAM_64 = 0xcffaedfe
+)
+
+var classBits = scalar.UToSymU{
+	MH_MAGIC:    32,
+	MH_CIGAM:    32,
+	MH_MAGIC_64: 64,
+	MH_CIGAM_64: 64,
+}
+
+var endianNames = scalar.UToSymStr{
+	MH_MAGIC:    "little_endian",
+	MH_CIGAM:    "big_endian",
+	MH_MAGIC_64: "little_endian",
+	MH_CIGAM_64: "big_endian",
+}
+
+var cpuTypes = scalar.SToSymStr{
```

Comment:
Maybe should sync these with elf somehow? or at least shorter and lowercase?

Reply:
I think it should be tackled in some other task

Reply:
👍